Mathlib/LinearAlgebra/SpecialLinearGroup.lean

```diff
@@ -0,0 +1,342 @@
+    Module.Free.ChooseBasisIndex.fintype R V
+  rw [← LinearMap.det_toMatrix b, ← LinearMap.det_toMatrix b.dualBasis]
+  rw [LinearMap.dualMap_def, LinearMap.toMatrix_transpose]
+  simp only [Matrix.det_transpose, LinearMap.det_toMatrix]
+
+/-- The transpose of an element in `SpecialLinearGroup R V`. -/
+def dualMap
+    [Module.Free R V] [Module.Finite R V] (A : SpecialLinearGroup R V) :
+    SpecialLinearGroup R (Module.Dual R V) :=
+  ⟨LinearEquiv.dualMap (A : V ≃ₗ[R] V), by
+    simp only [← Units.val_inj, LinearEquiv.coe_det, Units.val_one,
+      LinearEquiv.dualMap, LinearMap.det_dualMap]
+    simp [← LinearEquiv.coe_det, A.prop]⟩
+
+@[inherit_doc]
+scoped postfix:1024 "ᵀ" => SpecialLinearGroup.dualMap
+
+section CoeLemmas
+
+variable (A B : SpecialLinearGroup R V)
+
+theorem coe_mk (A : V ≃ₗ[R] V) (h : A.det = 1) : ↑(⟨A, h⟩ : SpecialLinearGroup R V) = A :=
+  rfl
+
+@[simp]
+theorem coe_mul : (A * B : SpecialLinearGroup R V) = (A * B  : V ≃ₗ[R] V) :=
+  rfl
+
+@[simp]
+theorem coe_div : (A / B : SpecialLinearGroup R V) = (A / B  : V ≃ₗ[R] V) :=
+  rfl
+
+@[simp]
+theorem coe_inv : (A : SpecialLinearGroup R V)⁻¹ = (A⁻¹ : V ≃ₗ[R] V) :=
+  rfl
+
+@[simp]
+theorem coe_one : (1 : SpecialLinearGroup R V) = (1 : V ≃ₗ[R] V) :=
+  rfl
+
+@[simp]
+theorem det_coe : LinearEquiv.det (A : V ≃ₗ[R] V) = 1 :=
+  A.prop
+
+@[simp]
+theorem coe_pow (m : ℕ) : (A ^ m : SpecialLinearGroup R V) = (A : V ≃ₗ[R] V) ^ m :=
+  rfl
+
+@[simp]
+theorem coe_zpow (m : ℤ) : (A ^ m : SpecialLinearGroup R V) = (A : V ≃ₗ[R] V) ^ m :=
+  rfl
+
+@[simp]
+theorem coe_dualMap
+    [Module.Free R V] [Module.Finite R V] :
+    Aᵀ = (A : V ≃ₗ[R] V).dualMap :=
+  rfl
+
+end CoeLemmas
+
+instance : Group (SpecialLinearGroup R V) :=
+  Function.Injective.group _ Subtype.coe_injective coe_one coe_mul coe_inv coe_div coe_pow coe_zpow
+
+/-- A version of `Matrix.toLin' A` that produces linear equivalences. -/
+def toLinearEquiv : SpecialLinearGroup R V →* V ≃ₗ[R] V where
+  toFun A := A.val
+  map_one' := coe_one
+  map_mul' := coe_mul
+
+theorem toLinearEquiv_apply (A : SpecialLinearGroup R V) (v : V) :
+    A.toLinearEquiv v = A v :=
+  rfl
+
+theorem toLinearEquiv_to_linearMap (A : SpecialLinearGroup R V) :
+    (SpecialLinearGroup.toLinearEquiv A) = (A : V →ₗ[R] V) :=
+  rfl
+
+theorem toLinearEquiv_symm_apply (A : SpecialLinearGroup R V) (v : V) :
+    A.toLinearEquiv.symm v = A⁻¹ v :=
+  rfl
+
+theorem toLinearEquiv_symm_to_linearMap (A : SpecialLinearGroup R V) :
+    A.toLinearEquiv.symm = ((A⁻¹ : SpecialLinearGroup R V) : V →ₗ[R] V) :=
+  rfl
+
+theorem toLinearEquiv_injective :
+    Function.Injective (toLinearEquiv : SpecialLinearGroup R V →* V ≃ₗ[R] V) :=
+  Subtype.val_injective
+
+section baseChange
+
+open TensorProduct
+
+variable {S : Type*} [CommRing S] [Algebra R S]
+  [Module.Free R V] [Module.Finite R V]
+
+/-- By base change, an `R`-algebra `S` induces a group homomorphism from
+`SpecialLinearGroup R V` to `SpecialLinearGroup S (S ⊗[R] V)`. -/
```

Comment:
```suggestion
`SpecialLinearGroup R V` to `SpecialLinearGroup S (S ⊗[R] V)`. -/
```
The white-space here is displayed in a weird way on my computer, I think it is some strange Unicode character.